src/stateMachines/maintenance/changePassword.ts

```diff
@@ -24,167 +21,247 @@ import { SecretManagerCreatorFactory } from '../../factories/SecretManagerCreato
 import { getPTStatusName, PTStatus } from '../../utils/PTStatus.js'
 import got from 'got'
 import { Environment } from '../../utils/Environment.js'
-
+import { invokeWsmanCall } from '../common.js'
+import { Error as ErrorStateMachine } from './../error.js'
 export interface SetAdminACLEntryExResponse {
-  SetAdminAclEntryEx_OUTPUT: {
-    ReturnValue: number
+  Envelope: {
+    Body: {
+      SetAdminAclEntryEx_OUTPUT: {
+        ReturnValue: number
+      }
+    }
   }
 }
 
 export const ChangePasswordEventType = 'CHANGE_PASSWORD'
-export type ChangePasswordEvent =
-  | { type: typeof ChangePasswordEventType, clientId: string, newStaticPassword: string }
+export interface ChangePasswordEvent {
+  type: typeof ChangePasswordEventType | 'ONFAILED'
+  clientId: string
+  newStaticPassword: string
+  output?: any
+}
 
 export interface ChangePasswordContext extends CommonMaintenanceContext {
-  newStaticPassword?: string
   generalSettings?: AMT.Models.GeneralSettings
+  newStaticPassword?: string
   updatedPassword?: string
 }
 
 const amt = new AMT.Messages()
 const logger = new Logger('changePassword')
 
 export class ChangePassword {
-  machine = createMachine<ChangePasswordContext, ChangePasswordEvent>({
-    id: 'change-password',
-    predictableActionArguments: true,
-    context: {
-      ...commonContext,
-      taskName: 'changepassword'
+  error: ErrorStateMachine = new ErrorStateMachine()
+  machine = setup({
+    types: {} as {
+      context: ChangePasswordContext
+      events: ChangePasswordEvent
+      input: ChangePasswordContext
+      actions: any
+    },
+    actors: {
+      getGeneralSettings: fromPromise(this.getGeneralSettings.bind(this)),
+      setAdminACLEntry: fromPromise(this.setAdminACLEntry.bind(this)),
+      saveToSecretProvider: fromPromise(this.saveToSecretProvider.bind(this)),
+      refreshMPS: fromPromise(this.refreshMPS.bind(this)),
+      error: this.error.machine
     },
+    guards: {
+      isGeneralSettings: ({ context, event }) => context.targetAfterError === 'GET_GENERAL_SETTINGS'
+    }
+  }).createMachine({
+    id: 'change-password',
+    context: ({ input }) => ({
+      clientId: input.clientId,
+      message: input.message,
+      errorMessage: input.errorMessage,
+      httpHandler: input.httpHandler,
+      taskName: input.taskName
+    }),
     initial: 'INITIAL',
     states: {
       INITIAL: {
         on: {
           CHANGE_PASSWORD: {
             actions: assign({
-              clientId: (context, event) => event.clientId,
-              newStaticPassword: (context, event) => event.newStaticPassword
+              clientId: ({ context, event }) => event.clientId,
+              newStaticPassword: ({ context, event }) => event.newStaticPassword
             }),
             target: 'GET_GENERAL_SETTINGS'
           }
         }
       },
       GET_GENERAL_SETTINGS: {
+        entry: assign({
+          message: ({ context, event }) => '',
```

Comment:
remove both un-used destructured elements and leave empty.  Apply this to all areas where context or event are not being read or used.